tests/integration/java/com/github/bazel_contrib/target_determinator/integration/BazelDifferIntegrationTest.java

```diff
@@ -53,7 +56,19 @@ private static void runProcess(Path workingDirectory, String... argv)
     // Do not clean the environment so we can inherit variables passed e.g. via --test_env.
     // Useful for CC (needed by bazel).
     processBuilder.environment().put("HOME", System.getProperty("user.home"));
-    processBuilder.environment().put("PATH", System.getenv("PATH"));
+
+    // There is a bug in `bazel-differ` where in `internal/bazel.go` they default to always using `bazel`
+    // rather than the the actual bazel instance we've requested. This is normally fine (just fine) but
+    // when we execute our own tests with `bazelisk`, the `PATH` has the version of `bazel` requested in
+    // our _own_ `.bazelversion` file at the head of the `PATH` (pulled from `bazelisk`'s own cache). To
+    // avoid this, we're going to remove items from the `PATH` that might be from bazelisk using a very
+    // sophisticated heuristic.
```

Comment:
```suggestion
    // rather than the the actual bazel instance we've requested. This is normally fine but when we
    // execute our own tests with `bazelisk`, the `PATH` has the version of `bazel` requested in our
    // _own_ `.bazelversion` file at the head of the `PATH` (pulled from `bazelisk`'s own cache). To
    // avoid this, we're going to remove items from the `PATH` that might be from bazelisk using a
    // best-effort heuristic.
```